src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -544,30 +555,105 @@ private Boolean isPassable(final BlockState blockState)
         return blockState.getMaterial() == Material.LEAVES;
     }
 
+
     /**
      * Check if distance to block changed and if we are not moving for too long, try to get unstuck.
      *
      * @return false
      */
-    private boolean checkIfStuckOnLeaves()
+    private boolean checkIfStuck()
     {
         if (!worker.getNavigator().noPath())
         {
-            Path pathToTree = worker.getNavigator().getPath();
-            if (pathToTree != null && pathToTree.getCurrentPathLength() > pathToTree.getCurrentPathIndex())
+            Path path = worker.getNavigator().getPath();
+            if (path != null)
             {
-                PathPoint next = pathToTree.getPathPointFromIndex(pathToTree.getCurrentPathIndex());
-                BlockPos nextPos = new BlockPos(next.x, next.y, next.z);
-                BlockPos nextPosUp = new BlockPos(next.x, next.y + 1, next.z);
-
-                if (world.getBlockState(nextPos).getBlock().isIn(BlockTags.LEAVES))
+                if (path.getCurrentPathLength() > path.getCurrentPathIndex())
                 {
-                    mineBlock(nextPos);
+                    return true;
                 }
-                else if (world.getBlockState(nextPosUp).getBlock().isIn(BlockTags.LEAVES))
+                if (path.getCurrentPathLength() == 0)
                 {
-                    mineBlock(nextPosUp);
+                    return true;
+                }
+            }
+            else
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Mines leaves on path and above.
+     */
+    private void tryUnstuck()
+    {
+        if (!worker.getNavigator().noPath())
+        {
+            Path path = worker.getNavigator().getPath();
+            if (path != null)
+            {
+                // Unstuck with path
+                ArrayList<BlockPos> checkPositions = new ArrayList<>();
+                PathPoint next = path.getPathPointFromIndex(path.getCurrentPathIndex());
+
+                // Blocks in front of the worker
+                for(int i = 0; i <= 2; i++){
```

Comment:
cough

Reply:
Done
i need to change the code style in intellj

Reply:
you can import the codestyle from the documentation folder